sdk/resourcemanager/azure-resourcemanager-cdn/src/main/java/com/azure/resourcemanager/cdn/implementation/CdnEndpointImpl.java

```diff
@@ -134,25 +132,34 @@ public String id() {
                 this.name(),
                 endpointUpdateParameters);
 
-        Mono<CustomDomainInner> customDomainTask = Flux.fromIterable(this.deletedCustomDomainList)
+        Flux<CustomDomainInner> customDomainCreateTask = Flux.fromIterable(this.customDomainList)
+            .flatMapDelayError(itemToCreate -> this.parent().manager().inner().getCustomDomains().createAsync(
+                this.parent().resourceGroupName(),
+                this.parent().name(),
+                this.name(),
+                self.parent().manager().sdkContext().randomResourceName("CustomDomain", 50),
+                itemToCreate.hostname()
+            ), 32, 32);
+
+        Flux<CustomDomainInner> customDomainDeleteTask = Flux.fromIterable(this.deletedCustomDomainList)
             .flatMapDelayError(itemToDelete -> this.parent().manager().inner().getCustomDomains().deleteAsync(
                 this.parent().resourceGroupName(),
                 this.parent().name(),
                 this.name(),
                 itemToDelete.name()
-            ), 32, 32)
-            .last();
+            ), 32, 32);
 
-        return Flux.zip(originUpdateTask, endpointUpdateTask, customDomainTask)
-            .last()
-            .map(objects -> {
-                self.setInner(objects.getT2());
+        Flux<CustomDomainInner> customDomainTask = Flux.concat(customDomainCreateTask, customDomainDeleteTask);
```

Comment:
If no order needed, use `Flux.mergeDelayError`

Reply:
The concern to have concat came from [Azure/azure-libraries-for-net#891](https://github.com/Azure/azure-libraries-for-net/issues/891). For safe, I just concat the custom domain tasks. For tasks of different resources, we can use merge.